antarest/study/storage/variantstudy/model/command_listener/command_listener.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2024, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+
+from abc import ABC, abstractmethod
+
+from antarest.core.serialization import AntaresBaseModel
+
+
+class ICommandListener(ABC, AntaresBaseModel):
+    """
+    Interface for all listeners that can be given inside the `apply` method of a command.
+    """
+
+    class Config:
+        extra = "forbid"
+        arbitrary_types_allowed = True
+
+    task_id: str
```

Comment:
We should not need this here anymore, once the command listener implementation will just forward to the task listener.

In the task service, we need to make an evolution of the `TaskJobLogRecorder` so that it records the progress to database and pushes progress to event bus, instead of doing it in the `TsGenerationListener` implementation.